packages/turf-point-grid/index.js

```diff
@@ -1,73 +1,72 @@
 import distance from '@turf/distance';
-import turfBBox from '@turf/bbox';
-import { point, featureCollection, isObject } from '@turf/helpers';
+import {point, featureCollection, isObject, isNumber} from '@turf/helpers';
 import inside from '@turf/inside';
-import { getType } from '@turf/invariant';
+import {getType} from '@turf/invariant';
 
 /**
  * Creates a {@link Point} grid from a bounding box, {@link FeatureCollection} or {@link Feature}.
  *
  * @name pointGrid
- * @param {BBox|GeoJSON} bbox extent in [minX, minY, maxX, maxY] order
- * @param {number} cellSide the distance between points
+ * @param {Array<number>} bbox extent in [minX, minY, maxX, maxY] order
+ * @param {number} cellSide the distance between points, in units
  * @param {Object} [options={}] Optional parameters
- * @param {string} [options.units="kilometers"] used in calculating cellSide, can be degrees, radians, miles, or kilometers
- * @param {number} [options.bboxIsMask=false] if true, and bbox is a Polygon or MultiPolygon, the grid Point will be created
+ * @param {string} [options.units='kilometers'] used in calculating cellSide, can be degrees, radians, miles, or kilometers
+ * @param {Feature<Polygon|MultiPolygon>} [options.mask] if passed a Polygon or MultiPolygon, the grid Points will be created only inside it
  * @param {Object} [options.properties={}] passed to each point of the grid
  * @returns {FeatureCollection<Point>} grid of points
  * @example
  * var extent = [-70.823364, -33.553984, -70.473175, -33.302986];
  * var cellSide = 3;
+ * var options = {units: 'miles'};
  *
- * var grid = turf.pointGrid(extent, cellSide, {units: 'miles'});
+ * var grid = turf.pointGrid(extent, cellSide, options);
  *
  * //addToMap
  * var addToMap = [grid];
  */
 function pointGrid(bbox, cellSide, options) {
     // Optional parameters
     options = options || {};
-    if (typeof options !== 'object') throw new Error('options is invalid');
-    var bboxIsMask = options.bboxIsMask || false;
-    var properties = options.properties || {};
+    if (!isObject(options)) throw new Error('options is invalid');
+    var units = options.units;
+    var maskIsPoly = options.mask && (getType(options.mask) === 'Polygon' || getType(options.mask) === 'MultiPolygon');
```

Comment:
I wouldn't use a `maskIsPoly` approach, instead add some type validation for the input and call it `mask`.